capstone-project/src/main/java/com/google/sps/servlets/MarkerServlet.java

```diff
@@ -73,6 +100,22 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     /** Stores a marker in Datastore. */
     public static void storeMarker(Marker marker) {
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        datastore.put(Marker.toEntity(marker));
+        Entity markerEntity = new Entity("Marker");
+        datastore.put(Marker.toEntity(marker, markerEntity));
     }
-}
+
+    /** Updates an existing marker's data */
+    public static void updateMarker(Marker marker) throws EntityNotFoundException {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Key markerEntityKey = KeyFactory.createKey("Marker", marker.getId());
+        Entity markerEntity = datastore.get(markerEntityKey);
```

Comment:
Why do we need to read the entity first? Don't we have all the updated data to store?

Reply:
In order to overwrite an existing Marker entity and not create a new one - I first found the relevant marker entity and then updated its properties. This way the entity's ID doesn't change and only the data fields are updated when we call the put() function. 
Did you mean that it'd be better if we sent the key to toEntity() and find the corresponding entity there instead of in this function?

Reply:
That's possible. 

You can have toEntity method that accepts ID as an argument (overloading the existing one.